### PR TITLE
GitHub actions on macos-latest

### PR DESCRIPTION
**Pull request by v1v**
Maybe #4 might affect this proposal?

### CODE REVIEW

**Comment by v1v** [comment]
@kvch , are you familiar with the below error?

```
env:
    GOROOT: /Users/runner/hostedtoolcache/go/1.17.8/x[6](https://github.com/elastic/elastic-agent-system-metrics/runs/5968534193?check_suite_focus=true#step:5:6)4
package github.com/elastic/elastic-agent-system-metrics: build constraints exclude all Go files in /Users/runner/work/elastic-agent-system-metrics/elastic-agent-system-metrics
Error: Process completed with exit code 1.
```

File: .github/workflows/macos.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      # Uses Go version from the repository.
+      - name: Read .go-version file
+        id: goversion
+        run: echo "::set-output name=version::$(cat .go-version)"
+
+      - uses: actions/setup-go@v2
+        with:
+          go-version: "${{ steps.goversion.outputs.version }}"
+
+      - name: Run build
```
Comment (kvch) [inline .github/workflows/macos.yml:25]: This is a library, you cannot run `go build` in this repo, just `go test`.

**Comment by kvch** [comment]
@fearful-symmetry Does the error on macos say anything to you?

**Comment by fearful-symmetry** [comment]
So, most of the MacOS metrics are legacy code, but I suspect that it's either a permissions issue, or we're trying to call `Statfs` on some kind of invalid filesystem type for that syscall. We might just want to exclude `vmhgfs` from the filesystem tests.

There's also this, which seems....interesting:
```
 numcpu_test.go:33: CPU Count: -1
```

**Comment by fearful-symmetry** [comment]
PR here to address some of the issues: https://github.com/elastic/elastic-agent-system-metrics/pull/13

**Review by kvch — APPROVED** [review]

**Comment by kvch** [comment]
Merging to solve golangci-lint issues in agent PR.

**Pull request merged**